lib/controllers/AssetController.ts

```diff
@@ -294,4 +291,25 @@ export class AssetController extends RelationalController {
     return super.delete(request);
   }
 
+  /**
+   * search assets or devices depending on the collection.
+   *
+   * @param request
+   */
+  async search (request: KuzzleRequest) {
+    const engineId = request.getString('engineId');
+    request.input.args.index = engineId;
+    const index = request.getIndex();
```

Comment:
Why do line 301 then 302 ? we can't just do use engineId directly ?

Reply:
Yes (I first try to use super.search() but after called directly this.sdk.document.search, this is not useful.